verification/curator-service/ui/src/components/AutomatedBackfill.tsx

```diff
@@ -0,0 +1,233 @@
+    caseReference?: CaseReferenceForm;
+    startDate: string | null;
+    endDate: string | null;
+}
+
+const AutomatedBackfillSchema = Yup.object().shape({
+    caseReference: Yup.object().shape({
+        sourceId: Yup.string().required('Required'),
+    }),
+    startDate: Yup.string().nullable().required('Required'),
+    endDate: Yup.string().nullable().required('Required'),
+});
+
+export default function AutomatedBackfill(props: Props): JSX.Element {
+    const classes = useStyles();
+    const [errorMessage, setErrorMessage] = React.useState('');
+    const [successMessage, setSuccessMessage] = React.useState('');
+
+    /**
+     * Convert the supplied date to a local YYYY-MM-DD format.
+     *
+     * Our date picker component returns a timezoned datetime, but for the
+     * purposes of backfill, users really just want to select a timezone-less
+     * date, as opposed to an instant in time.
+     *
+     * Converting to this format allows a proper range comparison for retrieval
+     * and parsing.
+     */
+    const isoDateString = (date: Date): string => {
+        const offsetMs = date.getTimezoneOffset() * 60 * 1000;
+        const msLocal = date.getTime() - offsetMs;
+        return new Date(msLocal).toISOString().split('T')[0];
+    };
+
+    const triggerBackfill = async (
+        values: AutomatedBackfillValues,
+    ): Promise<void> => {
+        try {
+            // This will never happen, because of Yup validation.
+            if (values.startDate === null || values.endDate === null) {
+                setErrorMessage('Please enter valid dates');
+                return;
+            }
+            const utcStartDateString = isoDateString(
+                new Date(values.startDate),
+            );
+            const utcEndDateString = isoDateString(new Date(values.endDate));
+            const baseUri = `/api/sources/${values.caseReference?.sourceId}/retrieve`;
+            const fullUri = `${baseUri}?parse_start_date=${utcStartDateString}&parse_end_date=${utcEndDateString}`;
+            const response = await axios.post<RetrievalResult>(
+                encodeURI(fullUri),
+            );
+            setErrorMessage('');
+            setSuccessMessage(`Created upload ${response.data.upload_id}.`);
+        } catch (e) {
+            setSuccessMessage('');
+            setErrorMessage(e.response?.data?.message || e.toString());
+            return;
+        }
+    };
+
+    return (
+        <AppModal
+            title="New automated source backfill"
+            onModalClose={props.onModalClose}
+        >
+            <Formik
+                validationSchema={AutomatedBackfillSchema}
+                validateOnChange={false}
+                initialValues={{
+                    caseReference: undefined,
+                    startDate: null,
+                    endDate: null,
+                }}
+                onSubmit={async (values): Promise<void> => {
+                    await triggerBackfill(values);
```

Comment:
if this returns a promise can you just return triggerBackfill(values)? instead of waiting for it?

Reply:
SG, done.